api_client/python/timesketch_api_client/sigma.py

```diff
@@ -95,3 +98,41 @@ def file_relpath(self):
             return ''
 
         return sigma_data.get('file_relpath', '')
+
+    def from_rule(self, rule_uuid):
+        """Get a Sigma object from a rule uuid.
+
+        Args:
+            rule_uuid: Id of the sigma rule.
+
+        """
+        # TODO: Write docstring
+        self.rule_uuid = rule_uuid
+        # TODO: not sure which one is the better one
+        self._resource_uri = f'sigma/{rule_uuid}'
+        self.resource_uri = f'sigma/{rule_uuid}'
+        super().__init__(
+            api=self.api, resource_uri=self._resource_uri)
```

Comment:
you should not do this... this is a `from_` function, not an `__init__` function

see how this is done in `search.py` for instance, that is the `__init__` function defines simply what variables are available, that is what you need to have saved...

then in the `from_rule_uuid` you set the resource_uri, and then you fetch the data for it, that is you do the fetching of data, if this is a GET request you can simple use `data = self.data` after you set the `self.resource_uri` and then you can start looking at the data...

it should have a `meta` and `objects` section, which you can then start to open up and assign variables, assign the `self._ATTR` values that you defined in `__init__`, then you've got properties that can fetch these values.

Reply:
but look at how this is done in `search.py` or `aggregation.py` for example.

you can see how the `from_*` functions operate there.

Essentially you take the input, make a request to teh API, interpret the results and save to variables.... then you also have `from_rule_text` or `from_something_else` and in those cases you may not need to do a request to the API, but just parse/interpret and assign to these same variables.

Reply:
IMHo it should now be much better, WDYT?